### PR TITLE
Spec changes 3.0 [docs]

### PR DESCRIPTION
**Pull request by phadej**
I changed how to build docs a little using `virtualenv`. That's IMHO a bit saner approach

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,6 +17,7 @@ register.sh
 cabal-tests.log
 
 /Cabal/dist/
+/Cabal/venv/
```
Comment (23Skidoo) [inline .gitignore]: Maybe call it something more obvious like `python-sphinx-virtualenv`?

File: Cabal/doc/file-format-changelog.rst [hunk 2]
```diff
@@ -32,6 +32,23 @@ relative to the respective preceding *published* version.
 * Remove deprecated ``hs-source-dir``, :pkg-field:`extensions` and
   :pkg-field:`build-tools` fields.
 
+* Common stanzas are allowed also in the beginnning of conditional sections.
```
Comment (23Skidoo) [inline Cabal/doc/file-format-changelog.rst]:
```suggestion
* Common stanzas are now allowed also in the beginnning of conditional sections.
```

File: Cabal/doc/file-format-changelog.rst [hunk 3]
```diff
@@ -32,6 +32,23 @@ relative to the respective preceding *published* version.
...
+  In other words following is valid
```
Comment (23Skidoo) [inline Cabal/doc/file-format-changelog.rst]:
```suggestion
  In other words, the following is valid
```

File: Cabal/doc/file-format-changelog.rst [hunk 4]
```diff
@@ -32,6 +32,23 @@ relative to the respective preceding *published* version.
...
+    ::
+
+        library
+            import deps
+
+            if flag(foo)
+                import foo-deps
+
+* Allow redundant leading or trailing commas in package fields with
+  optional commas such as :pkg-field:`exposed-modules`
```
Comment (23Skidoo) [inline Cabal/doc/file-format-changelog.rst]:
```suggestion
  optional commas, such as :pkg-field:`exposed-modules`.
```

**Comment by phadej** [comment]
I made suggested changes, merging

**Pull request merged**